src/escape.rs

```diff
@@ -31,6 +31,13 @@ pub struct EscapePass;
 /// ```
 declare_lint!(pub BOXED_LOCAL, Warn, "using Box<T> where unnecessary");
 
+fn is_box(ty: ty::Ty) -> bool {
```

Comment:
I extracted it to a function because 1. it is used in two places (for locals and for arguments) 2. we'd want it to match String and Vec in the future.